tests/test_planner_output/test_command.py

```diff
@@ -0,0 +1,361 @@
+    assert complex_path.joint_names == [f"joint_{num}" for num in range(1, 8)]
+    assert complex_path.dimensionality == 7
+    assert complex_path.num_waypoints == 11
+
+
+def test_joint_space_path_waypoints_for_joint(complex_path):
+    """Getting the waypoints for a given joint"""
+    assert (
+        complex_path.waypoints_for_joint("joint_5") == complex_path.waypoints["joint_5"]
+    )
+    with pytest.raises(ValueError):
+        complex_path.waypoints_for_joint("joint_999")
+
+
+def test_joint_space_path_waypoint(complex_path):
+    """Test getting the waypoints at a given index"""
+    assert complex_path.waypoint(0) == _expected_waypoint_at_idx(
+        complex_path.waypoints, 0
+    )
+    assert complex_path.waypoint(10) == _expected_waypoint_at_idx(
+        complex_path.waypoints, 10
+    )
+    # Check we can do negative indexing
+    assert complex_path.waypoint(10) == complex_path.waypoint(-1)
+    assert complex_path.waypoint(-11) == complex_path.waypoint(0)
+
+    with pytest.raises(ValueError):
+        complex_path.waypoint(999)
+
+
+def test_joint_space_path_waypoint_as_np_array(complex_path):
+    """Testing getting the waypoints at a given index as a np.array"""
+    # joint_1, joint_2, ..., joint_7
+    joint_name_ordering = [f"joint_{num}" for num in range(1, 8)]
+    expected_waypoint_at_idx_0 = list(
+        _expected_waypoint_at_idx(complex_path.waypoints, 0).values()
+    )
+    expected_waypoint_at_idx_10 = list(
+        _expected_waypoint_at_idx(complex_path.waypoints, 10).values()
+    )
+
+    assert np.array_equal(
+        complex_path.waypoint_as_np_array(0, joint_name_ordering),
+        np.array(expected_waypoint_at_idx_0),
+    )
+    assert np.array_equal(
+        complex_path.waypoint_as_np_array(10, joint_name_ordering),
+        np.array(expected_waypoint_at_idx_10),
+    )
+
+    # Reverse ordering: joint_7, joint_6, ..., joint_1
+    reversed_joint_name_ordering = list(reversed(joint_name_ordering))
+    assert np.array_equal(
+        complex_path.waypoint_as_np_array(0, reversed_joint_name_ordering),
+        np.array(list(reversed(expected_waypoint_at_idx_0))),
+    )
+    assert np.array_equal(
+        complex_path.waypoint_as_np_array(10, reversed_joint_name_ordering),
+        np.array(list(reversed(expected_waypoint_at_idx_10))),
+    )
+
+    # Invalid joint name ordering 1_joint, 2_joint, etc.
+    with pytest.raises(ValueError):
+        complex_path.waypoint_as_np_array(
+            0, [f"{idx + 1}_joint" for idx in range(1, 8)]
+        )
+
+
+def test_joint_space_paths_waypoints_as_np_array(complex_path):
+    """Test getting all the waypoints as a np.array"""
+    # joint_1, joint_2, ..., joint_7
+    joint_name_ordering = [f"joint_{num}" for num in range(1, 8)]
+    expected_np_array = np.array(
+        [complex_path.waypoints[joint_name] for joint_name in joint_name_ordering]
+    )
+    assert expected_np_array.shape == (7, 11)
```

Comment:
Is this the right shape we want to return or transpose?

Reply:
We want transpose IMO. Not a huge detail as different sims have different formatting preferences I imagine but if I have some array of data X (and waypoints as np array is data containing waypoints), then each row should contain a waypoint with columns corresponing to joints for the waypoint. Also waypoint_as_np_array and waypoints_as_np_array should have a comment mentioning the shape of the output.

Reply:
Yep sounds good